db/migrate/20150223125700_create_branches.rb

```diff
@@ -0,0 +1,15 @@
+class CreateBranches < ActiveRecord::Migration
+  def up
+    create_table(:branches) do |t|
+      t.integer :repository_id, null: false
+      t.integer :last_build_id
+      t.string  :name
+      t.timestamps
+    end
+    add_index(:branches, [:repository_id, :name])
```

Comment:
Good catch. Will that error out if we happen to have two branch objects with the same name?